### ISSUE TITLE
Do I have to do anything special to use BatchNormalization in my own training loops?

### ISSUE THREAD

**Issue by isaacgerg** [keras-team/keras#13316]
Using tf 1.13.1 and tf 1.15.  I have to have a loop to do my own training (e.g. i call model.train() to train my model). Do I have to do anything other special when using BatchNormalizationi n my model when training like this? I am assuming it takes care of handling the training/test state internally.  Is this correct?

**EDIT 1**: So it turns out that the keras source does not update the variables for the EMA unless you are using model.fit().  Using model.train_on_batch() will result in the EMA variables not being updated.  The keras documentation should specifically mention this.

**Comment by KazukiUruma** [comment]
I don't think you have to do nothing special when you train your model with BatchNormalization Layers.
When you call model.train(), BN Layer learn its params, and when you call model.predict(), BN Layer use its learned param. This is done automatically.

**Comment by isaacgerg** [comment]
@KazukiUruma I checked the source code and am not seeing what you mentioned. When I call train, nothing in keras collects batchnorm variables so they are never trained.  I have verified this experimentally also.

**Comment by isaacgerg** [comment]
I have narrowed the problem down more.  Specifically, this is the same problem as this: #9522 .

Say I have a given tensor i want to classify called x.  I get a certain y for y = model.predict(x).  If i augument x with additional data tensors to make a larger batch size x4 = x + [x2,x3, x4] (say 4 in this case now), I dont get the same prediction for x.
y = model.predict(x)
y4 = model.predict(x4)
y[0] should equal y4[0] but this is not the case. Its off more than precision error also.  (in my case 0.3 for a sigmoid last layer output)

**Comment by isaacgerg** [comment]
Here is code to reproduce the issue:

```
    input1 = tf.keras.layers.Input(shape=(128,128,1))
    x = tf.keras.layers.BatchNormalization()(input1)
    x = tf.keras.layers.Flatten()(x)
    x = tf.keras.layers.Dense(5, activation = 'sigmoid')(x)
    
    model = tf.keras.models.Model(input1, x)

    batchSize = 32
    x = np.random.rand(batchSize, 128,128,1)
    
    print('The following rows should all be equal...')
    for k in range(1, batchSize):  
        y = model.predict(x[0:k,:,:,:])
        print(y[0,:])
```
